### PR TITLE
adding support to filter on 'vpc attribute'

### PR DESCRIPTION
**Pull request by joshuaroot**
Per #1616 , adding support to filter VPC based on vpc-attribute

### CODE REVIEW

File: c7n/resources/vpc.py [hunk 1]
```diff
@@ -191,6 +191,34 @@ def get_related_ids(self, resources):
...
+        EnableDnsSupport={'type': 'boolean'})
+    permissions = ('ec2:DescribeVpcAttributes',)
+
+    def process(self, resources, event=None):
+        results = []
+        client = local_session(self.manager.session_factory).client('ec2')
+
+        for r in resources:
+            hostname = client.describe_vpc_attribute(
+                VpcId=r['VpcId'],
```
Comment (kapilt) [inline c7n/resources/vpc.py]: Ideally only fetch the attribute if it’s being matched on

File: c7n/resources/vpc.py [hunk 2]
```diff
@@ -191,6 +191,34 @@ def get_related_ids(self, resources):
         return vpc_group_ids
 
 
+@Vpc.filter_registry.register('vpc-attributes')
+class AttributesFilter(Filter):
+    """Filters VPCs based on their DNS attributes"""
+    schema = type_schema(
+        'vpc-attributes',
+        EnableDnsHostnames={'type': 'boolean'},
```
Comment (kapilt) [inline c7n/resources/vpc.py]: We be generally gone for lower case hyphenated when not letting the user specify arbitrary keys

Comment (joshuaroot) [reply c7n/resources/vpc.py]: gotcha. I had originally used lower case for the parameters but noted a few instances where upper case was used.

**Comment by joshuaroot** [comment]
I also forgot to add in a mock policy in the docstring...

**Comment by kapilt** [comment]
i'd suggest we also drop the 'enable' prefix on the attributes.

**Pull request merged**